### DISCUSSION TITLE
Verifying Spegel is working

### DISCUSSION THREAD

**Discussion by bojanraic** [k3s-io/k3s#10071] [Q&A] [answered]
I have a cluster with Spegel enabled via the config file. I can see ports are open and every node can reach other nodes. 
However, when I pull an image for a pod scheduled on one node, it takes, say, 5 minutes. When I schedule the same pod on a different node, I do not see any speed difference in pulling the image from the peer.
My question is, how can I test and verify that Spegel is working as intended? 
Thanks!

**Comment by brandond** [comment, marked as answer]
Check the metrics?

`kubectl get --raw /api/v1/nodes/<NODENAME>/proxy/metrics | grep -F 'spegel'`

Are you sure you actually enabled mirroring for the upstream registry as covered in the docs?
https://docs.k3s.io/installation/registry-mirror#enabling-registry-mirroring

**Reply by bojanraic** [reply]
> Check the metrics?
> 
> 
> 
> `kubectl get --raw /api/v1/nodes/<NODENAME>/proxy/metrics | grep -F 'spegel'`
> 
> 
> 
> Are you sure you actually enabled mirroring for the upstream registry as covered in the docs?
> 
> https://docs.k3s.io/installation/registry-mirror#enabling-registry-mirroring

Thanks for that! I can see cache hits on some of the nodes. 
Cheers!
